ui/src/AccountDropdown/InviteEditorsFormSection.tsx

```diff
@@ -43,29 +69,84 @@ const getUsers = (currentSpace: Space, setUsersList: (usersList: UserSpaceMappin
     }
 };
 
-function InviteEditorsFormSection({collapsed, currentSpace, currentUser, closeModal, setCurrentModal}: Props): JSX.Element {
+function InviteEditorsFormSection({collapsed, currentSpace, currentUser, closeModal, setCurrentModal}: InviteEditorsFormProps): JSX.Element {
     const isExpanded = !collapsed;
     const [invitedUserEmails, setInvitedUserEmails] = useState<string[]>([]);
+    const [invitedUserIds, setInvitedUserIds] = useState<Option[]>([]);
+    const [inputValue, setInputValue] = useState<string>('');
     const [enableInviteButton, setEnableInviteButton] = useState<boolean>(false);
     const [usersList, setUsersList] = useState<UserSpaceMapping[]>([]);
+    const [showErrorMessage, setShowErrorMessage] = useState<boolean>(false);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const components: any = {
+        DropdownIndicator: null,
+    };
+
+    const FEATURE_TOGGLE = window.location.hash === '#perm';
 
     useEffect(() => {
         getUsers(currentSpace, setUsersList);
     }, [currentSpace, setUsersList]);
 
     const inviteUsers = async (event: FormEvent): Promise<void> => {
         event.preventDefault();
+        if (FEATURE_TOGGLE ) {
+            await SpaceClient.inviteUsersToSpace(currentSpace, invitedUserIds.map(userId => userId.value))
+                .catch(console.error)
+                .finally(() => {
+                    setCurrentModal({modal: AvailableModals.GRANT_EDIT_ACCESS_CONFIRMATION});
+                });
+        } else {
+            await SpaceClient.oldInviteUsersToSpace(currentSpace, invitedUserEmails)
+                .catch(console.error)
+                .finally(() => {
+                    setCurrentModal({modal: AvailableModals.GRANT_EDIT_ACCESS_CONFIRMATION});
+                });
+        }
+    };
+
+    useEffect(() => {
+        const enable = (invitedUserIds.length > 0 && inputValue.trim().length === 0)
+                || (!!inputValue.trim().match(userIdPattern))
+                || (invitedUserEmails.length > 0);
```

Comment:
Woof. This is an ugly conditional

Reply:
To do: Add TODO comments in code to identify code that can be removed when we make change this code from feature toggled to live. (Removing the email entries)